### DISCUSSION TITLE
Installed driver on cerbo but doesnt seem to be working

### DISCUSSION THREAD

**Discussion by johnnyboy70** [Louisvdw/dbus-serialbattery#420] [General]
Im not sure what ive done wrong but no usb is showing up when i try the troubleshooting commands . Any ideas ?
John

~~~bash
Using username "root".
root@venus.local's password:
Last login: Sun Jan 15 19:31:38 2023
root@nanopi:~# tail -f /data/log/dbus-serialbattery.ttyUSB0/current
tail: can't open '/data/log/dbus-serialbattery.ttyUSB0/current': No such file or directory
tail: no files
root@nanopi:~# tail -f /data/log/serial-starter/current
@4000000063c454a00e845cac *** starting serial-starter ***
@4000000063c454a021ce3684 serstart starting
@4000000063c454a022a9b9d4 INFO: loading config file /etc/venus/serial-starter.conf
@4000000063c454a10c13cdf4 INFO: loading config file /data/conf/serial-starter.d
@4000000063c454a206769c14 INFO: Create daemontools service vedirect-interface.ttyS1
@4000000063c454a322957e24 INFO: Create daemontools service vedirect-interface.ttyS2
@4000000063c454a502241a74 INFO: Create daemontools service mk2-dbus.ttyS3
@4000000063c454a812db3b7c INFO: Start service vedirect-interface.ttyS1
@4000000063c454a927d578bc INFO: Start service vedirect-interface.ttyS2
@4000000063c454ab101d3c8c INFO: Start service mk2-dbus.ttyS3
~~~

**Comment by johnnyboy70** [comment]
root@nanopi:~# /data/log/serial-starter/current
-sh: /data/log/serial-starter/current: Permission denied

**Comment by Louisvdw** [comment]
Check that the driver did install on your device by running 
>ls /data/etc/dbus-serialbattery/

If you get a error that the directory does not exist, then redo the installation.

Otherwise reboot your device twice and see if it picks up then. 
You have to plug something in to the USB port on your GX device before it will check for drivers for that device.

**Reply by johnnyboy70** [reply]
Hi Louis , thanks for the reply . Looks like its installed ? Rebooted twice but the JK bms still isnt showing  . Im using the adaptors that Andy @ offgridgarage recommended so they should be ok ?

root@nanopi:~# ls /data/etc/dbus-serialbattery/
LICENSE                 installlocal.sh         reinstalllocal.sh
README.md               installqml.sh           renogy.py
ant.py                  installrelease.sh       restoregui.sh
battery.py              jkbms.py                service
daly.py                 lifepower.py            sinowealth.py
dbus-serialbattery.py   lltjbd.py               start-serialbattery.sh
dbushelper.py           minimalmodbus.py        util_max17853.py
disabledriver.sh        mnb.py                  utils.py
ecs.py

**Reply by Louisvdw** [reply]
Those adapters should work and the driver does seem to be installed. 
If you plug something into the USB port the log file for the serial-starter should show activity by starting all the drivers it knows about and the serialbattery should be one of them. Check if that happens or not. 
If serial-starter does show the seriabattery is being start look at the corresponding log file to see what is happening. 

Also check your wiring and for any loose connection. And check that your adapter does work

**Reply by johnnyboy70** [reply]
Sorry Louis , im new to this and dont really know what im doing . Ive tried two different RS485 adaptors , the waveshare  (https://amzn.eu/d/dJKjqyF) has a power light on . I dont know how to test the Jikong BMS RS485 Adapter . 

I rebooted and tried these commands again : 

~~~bash
root@nanopi:~# /data/log/serial-starter/current
-sh: /data/log/serial-starter/current: Permission denied
root@nanopi:~# /data/log/dbus-serialbattery.ttyUSB0/current
-sh: /data/log/dbus-serialbattery.ttyUSB0/current: No such file or directory
root@nanopi:~# /data/log/dbus-serialbattery.ttyUSB1/current
-sh: /data/log/dbus-serialbattery.ttyUSB1/current: No such file or directory
root@nanopi:~# /data/log/dbus-serialbattery.ttyUSB2/current
-sh: /data/log/dbus-serialbattery.ttyUSB2/current: No such file or directory
root@nanopi:~# /data/log/dbus-serialbattery.ttyUSB3/current
-sh: /data/log/dbus-serialbattery.ttyUSB3/current: No such file or directory
~~~

Apologies if im being stupid ...

**Reply by Louisvdw** [reply]
You are trying to execute that file, which is why it gives permission error. Use it like
> tail -f /data/log/serial-starter/current

Ctrl-C to stop.

Does the adapters have lights that blink? Which ones and only once or frequent? Can you share a picture of how you connected it?

**Reply by johnnyboy70** [reply]
The transmit and receive lights are not blinking , just the power light is on .

![20230116_210228](https://user-images.githubusercontent.com/31316937/212765682-458bc3bb-8c0b-49f1-9f64-99cf95450a6f.jpg)

**Comment by mr-manuel** [comment]
Did you solve your problem?

**Reply by johnnyboy70** [reply]
No , there seems to be no signal coming from the bms . I decided to get a smart shunt .
